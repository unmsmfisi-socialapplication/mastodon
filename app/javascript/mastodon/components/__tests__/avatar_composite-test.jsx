import { fromJS } from 'immutable';

import renderer from 'react-test-renderer';

import AvatarComposite from '../avatar_composite';

describe('<AvatarComposite />', () => {
  const accounts = fromJS([
    { id: '1', username: 'alice', acct: 'alice@domain.com', display_name: 'Alice', avatar: '/static/alice.jpg' },
    { id: '2', username: 'bob', acct: 'bob@domain.com', display_name: 'Bob', avatar: '/static/bob.jpg' },
    { id: '3', username: 'charlie', acct: 'charlie@domain.com', display_name: 'Charlie', avatar: '/static/charlie.jpg' },
    { id: '4', username: 'dave', acct: 'dave@domain.com', display_name: 'Dave', avatar: '/static/dave.jpg' },
  ]);

  it('renders correctly with a list of accounts and size 100', () => {
    const component = renderer.create(
      <AvatarComposite accounts={accounts} size={100} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with more than 4 accounts', () => {
    const accountsExtended = fromJS([
      ...accounts.toJS(),
      { id: '5', username: 'eve', acct: 'eve@domain.com', display_name: 'Eve', avatar: '/static/eve.jpg' },
      { id: '6', username: 'frank', acct: 'frank@domain.com', display_name: 'Frank', avatar: '/static/frank.jpg' },
    ]);
    const component = renderer.create(
      <AvatarComposite accounts={accountsExtended} size={100} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with animation disabled', () => {
    const component = renderer.create(
      <AvatarComposite accounts={accounts} size={100} animate={false} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with less than 4 accounts', () => {
    const fewAccounts = fromJS([
      { id: '1', username: 'alice', acct: 'alice@domain.com', display_name: 'Alice', avatar: '/static/alice.jpg' },
      { id: '2', username: 'bob', acct: 'bob@domain.com', display_name: 'Bob', avatar: '/static/bob.jpg' },
    ]);
    const component = renderer.create(
      <AvatarComposite accounts={fewAccounts} size={100} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
