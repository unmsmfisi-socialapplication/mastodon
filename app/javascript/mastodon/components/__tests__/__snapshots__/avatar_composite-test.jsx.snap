// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarComposite /> renders correctly with a list of accounts and size 100 1`] = `
<div
  className="account__avatar-composite"
  style={
    {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div
    style={
      {
        "bottom": "1px",
        "height": "50%",
        "left": "auto",
        "right": "1px",
        "top": "auto",
        "width": "50%",
      }
    }
  >
    <div
      className="account__avatar account__avatar--loading"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        {
          "height": "20px",
          "width": "20px",
        }
      }
    />
  </div>
  <div
    style={
      {
        "bottom": "1px",
        "height": "50%",
        "left": "1px",
        "right": "auto",
        "top": "auto",
        "width": "50%",
      }
    }
  >
    <div
      className="account__avatar account__avatar--loading"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        {
          "height": "20px",
          "width": "20px",
        }
      }
    />
  </div>
  <div
    style={
      {
        "bottom": "auto",
        "height": "50%",
        "left": "auto",
        "right": "1px",
        "top": "1px",
        "width": "50%",
      }
    }
  >
    <div
      className="account__avatar account__avatar--loading"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        {
          "height": "20px",
          "width": "20px",
        }
      }
    />
  </div>
  <div
    style={
      {
        "bottom": "auto",
        "height": "50%",
        "left": "1px",
        "right": "auto",
        "top": "1px",
        "width": "50%",
      }
    }
  >
    <div
      className="account__avatar account__avatar--loading"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        {
          "height": "20px",
          "width": "20px",
        }
      }
    />
  </div>
</div>
`;

exports[`<AvatarComposite /> renders correctly with more than 4 accounts 1`] = `
<div
  className="account__avatar-composite"
  style={
    {
      "height": "100px",
      "width": "100px",
    }
  }
>
  <div
    style={
      {
        "bottom": "1px",
        "height": "50%",
        "left": "auto",
        "right": "1px",
        "top": "auto",
        "width": "50%",
      }
    }
  >
    <div
      className="account__avatar account__avatar--loading"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        {
          "height": "20px",
          "width": "20px",
        }
      }
    />
  </div>
  <div
    style={
      {
        "bottom": "1px",
        "height": "50%",
        "left": "1px",
        "right": "auto",
        "top": "auto",
        "width": "50%",
      }
    }
  >
    <div
      className="account__avatar account__avatar--loading"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        {
          "height": "20px",
          "width": "20px",
        }
      }
    />
  </div>
  <div
    style={
      {
        "bottom": "auto",
        "height": "50%",
        "left": "auto",
        "right": "1px",
        "top": "1px",
        "width": "50%",
      }
    }
  >
    <div
      className="account__avatar account__avatar--loading"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        {
          "height": "20px",
          "width": "20px",
        }
      }
    />
  </div>
  <div
    style={
      {
        "bottom": "auto",
        "height": "50%",
        "left": "1px",
        "right": "auto",
        "top": "1px",
        "width": "50%",
      }
    }
  >
    <div
      className="account__avatar account__avatar--loading"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        {
          "height": "20px",
          "width": "20px",
        }
      }
    />
  </div>
  <span
    className="account__avatar-composite__label"
  >
    +
    2
  </span>
</div>
`;
